spring-integration-cloudevents/src/main/java/org/springframework/integration/cloudevents/transformer/ToCloudEventTransformer.java

```diff
@@ -0,0 +1,415 @@
+	 * @since 7.1
+	 */
+	public void setTypeExpression(Expression typeExpression) {
+		this.typeExpression = typeExpression;
+	}
+
+	/**
+	 * Set the {@link Expression} to create the dataSchema for the CloudEvent.
+	 * @param dataSchemaExpression the expression to create the dataSchema for each CloudEvent
+	 * @since 7.1
+	 */
+	public void setDataSchemaExpression(Expression dataSchemaExpression) {
+		this.dataSchemaExpression = dataSchemaExpression;
+	}
+
+	/**
+	 * Set the {@link Expression} to create the subject for the CloudEvent.
+	 * @param subjectExpression the expression to create the subject for each CloudEvent
+	 * @since 7.1
+	 */
+	public void setSubjectExpression(Expression subjectExpression) {
+		this.subjectExpression = subjectExpression;
+	}
+
+	/**
+	 * Set to {@code true} to fail if no {@link EventFormat} is found for message content type.
+	 * When {@code false} and no {@link EventFormat} is found, then a {@link CloudEvent}' body is
+	 * set as an output message's payload, and its attributes are set into headers.
+	 * @param failOnNoFormat true to disable format serialization
+	 * @since 7.1
+	 */
+	public void setFailOnNoFormat(boolean failOnNoFormat) {
+		this.failOnNoFormat = failOnNoFormat;
+	}
+
+	/**
+	 * Return whether the transformer will transform the message
+	 * when no {@link EventFormat} is available for the content type.
+	 * @return {@code true} if transformation should fail
+	 *         when no suitable {@link EventFormat} is found;
+	 *         {@code false} otherwise
+	 * @since 7.1
+	 */
+	public boolean isFailOnNoFormat() {
+		return this.failOnNoFormat;
+	}
+
+	/**
+	 * Set the prefix for CloudEvent headers in binary content mode.
+	 * @param cloudEventPrefix the prefix to use for CloudEvent headers
+	 * @since 7.1
+	 */
+	public void setCloudEventPrefix(String cloudEventPrefix) {
+		this.cloudEventPrefix = cloudEventPrefix;
+	}
+
+	/**
+	 * Return the prefix used for CloudEvent headers in binary content mode.
+	 * @return the CloudEvent header prefix
+	 * @since 7.1
+	 */
+	public String getCloudEventPrefix() {
+		return this.cloudEventPrefix;
+	}
+
+	/**
+	 * Transform the input message into a CloudEvent message.
+	 * @param message the input Spring Integration message to transform
+	 * @return CloudEvent message in the specified format
+	 * @throws RuntimeException if serialization fails
+	 */
+	@Override
+	protected Object doTransform(Message<?> message) {
+		Assert.isInstanceOf(byte[].class, message.getPayload(), "Message payload must be byte[]");
+
+		String id = this.eventIdExpression.getValue(this.evaluationContext, message, String.class);
+		URI source = this.sourceExpression.getValue(this.evaluationContext, message, URI.class);
+		String type = this.typeExpression.getValue(this.evaluationContext, message, String.class);
+		MessageHeaders headers = message.getHeaders();
+		MessageHeaderAccessor accessor = new MessageHeaderAccessor(message);
+		MimeType mimeType = accessor.getContentType();
```

Comment:
Please, consider to use `StaticMessageHeaderAccessor.getContentType()` instead.
That way we won't have an extra `MessageHeaderAccessor` volatile object.